kong/api/endpoints.lua

```diff
@@ -28,10 +32,79 @@ local ERRORS_HTTP_CODES = {
 }
 
 
+local function get_message(default, ...)
+  local message
+  local n = select("#", ...)
+  if n > 0 then
+    if n == 1 then
+      local arg = select(1, ...)
+      if type(arg) == "table" then
+        message = arg
+      elseif arg ~= nil then
+        message = tostring(arg)
+      end
+
+    else
+      message = {}
+      for i = 1, n do
+        local arg = select(i, ...)
+        message[i] = tostring(arg)
+      end
+      message = concat(message)
+    end
+  end
+
+  if not message then
+    message = default
+  end
+
+  if type(message) == "string" then
+    message = { message = message }
+  end
+
+  return message
+end
+
+
+local function ok(...)
```

Comment:
Please add the params explicitly. We shouldn't have to look at a function's implementation in order to know which params it needs.

Reply:
You cannot, as it uses varargs.

Reply:
> My suggestion would be splitting things differently. Move everything self-related from query_entity into extract_options. Then export extract_options, and not update_entity etc.

A lot of custom endpoints already utilized `select_entity`. There are several things that endpoints export:

1. normal api generators: e.g. get_entity_endpoint, put_entity_endpoint etc. that can be modified by providing a method argument. This is highest level of abstraction. But sometimes you need more specific ones:

2. database query generators with parameter parsing and handling of function return codes: select_entity, update_entity etc. (this can be seen as a bit like a crud helpers, and soon I hope @hbagdi's resolving of unique fields to primary keys will be added for update, upsert, and insert).

3. smaller helpers: extract_options that returns database call `options` table (adds ttl support and nulls option to all admin api db calls, whenever you remember to use that, but 1. and 2. always does).

It is always suggested to use highest possible abstraction, and if that does not give answer to your problem, use lower, and use even normal kong.db with 3. But as you move lesser abstractions, there is a great change that you make mistakes here and there, that leads to inconsistent admin apis.

Reply:
I can go and refactor the `2.`, e.g. remove `self`, and `db`, and possibly add `params` (the uri matches) and `args` (the results that arguments.lua returns).